### ISSUE TITLE
Implement page performance checks for ursus-dev

### ISSUE THREAD

**Issue by mark-dce** [UCLALibrary/californica#246]
**ACCEPTANCE**
- [x] Implement the checks determined in #111 for the ursus-dev server using StatusCake
- [x] Documenting Homepage load time
- [x] Documenting Page 3 of search results e.g. https://californica-test.library.ucla.edu/catalog?locale=en&page=3&q=japanese&search_field=all_fields load time
- [x] Documenting Single Item view (e.g. https://californica-test.library.ucla.edu/concern/works/k930bx013 load time including how long it takes for the image to go from fuzzy gray pixels to an actual discernible image

**Comment by mark-dce** [comment]
Assumes we're using StatusCake to be consistent with other UCLA DIIIT web service monitoring.

**Comment by emcaulay** [comment]
Is status cake effective for the single item loading? For example,
this page loads pretty quickly, but the image takes forever, and single item view load includes completing https://californica-test.library.ucla.edu/concern/works/sb397842d?locale=en#?c=0&m=0&s=0&cv=0&xywh=-3270%2C-88%2C12739%2C5444. 
See also: https://screencast-o-matic.com/watch/cF6o3QYtPL

**Comment by ksclarke** [comment]
For what it's worth, we might need to look into how StatusCake measures "done". For instance, an item's image, when it's in a OpenSeadragon viewer, loads _after_ the page finishes loading. It's all done through Ajax, so something that's measuring when the page has loaded isn't necessarily measuring when all the parts of the page (e.g., tiled images) have finished loading.

**Comment by emcaulay** [comment]
Yes, what @ksclarke is explaining is the concern I had.

**Comment by mark-dce** [comment]
I'm not sure how to turn this concern into an actionable ticket other than by implementing StatusCake and observing that data it can provide.

If there is a faster way to figure this out than enabling the StatusCake checks and seeing what data they provide, let's lay out what that is and do that instead; if not, let's put the StatusCake checks in place and figure out where the gaps are.

**Comment by mark-dce** [comment]
FYI - As a baseline, ad-hoc testing shows it takes anywhere from 19 to 34s for the complete page to load.  We can always continue doing ad-hoc testing in this way to supplement StatusCake, but I think it would be a mistake to forego using StatusCake since it's available to us.

<img width="1436" alt="as-124 fish harbor load time" src="https://user-images.githubusercontent.com/3064318/46992779-995b5d00-d0c0-11e8-9937-004603c70561.png">


<img width="1437" alt="as-124 myrtle l mantz load time" src="https://user-images.githubusercontent.com/3064318/46992780-995b5d00-d0c0-11e8-8fa2-dd5560691003.png">

**Comment by ksclarke** [comment]
I don't think anyone is saying not to use status cake. We just need to know what it's measuring (we want to make sure we're answering the questions that we're asking).

Actions:
 * Find out what statuscake is measuring (page load or loading of all content on the page)
 * If it's measuring the loading of all content on a page, we're good to go using it as a tool for all our acceptance criteria on this ticket and we do that
 * If it's measuring page load only then we use it for things that don't display images using Ajax (all the acceptance criteria except the last one)

If it can't measure Ajax image loading we can move acceptance criteria number four to another ticket and look for a tool that will answer that question for us.

**Comment by ksclarke** [comment]
For what it's worth, we do have a tool that requests all the images OpenSeadragon would request from a IIIF server and times how long the server takes to deliver them. That's not directly answering the question either because we're interested in the user experience and that tool doesn't measure putting those images on an HTML page, but it's better than ad hoc testing for getting some basic metrics that we can use to measure improvements/changes, I think.

**Comment by cachemeoutside** [comment]
I've added statuscake pagespeed checks on the 3 links that Mark provided in the acceptance criteria. At this point, I'm unsure how to share these results. That'll be the next thing I'll look into if it's desired. But here's a few quick screenshots to show the detail of the pagespeed checks.
![singleitemview.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5b3fa152acf9c70015453b07/125516c66e82c728ace21e0d46fb9d94248e86a09becb255ad58e7d665043761260ea62864c7b96aea4028040b5e4aa448095f4dadac892ba7e43a798d5a0ae8c3714d36976345f3e5db55ea43a82a994e8dbcbd8f48c0ed3526ef8e8d4a391b1755.png)
![searchresults.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5b3fa152acf9c70015453b07/125516c66e82c728ace21e0d46fb919b31818bbb8afaaa4fb04eb2d86b56357c2301a03f32dfa17dbd117e055e5b46a510585615a9f9892af4b035778c5d0effcd7f1d2598694ef1e1c84afd13b0328a468fb7b98e4cc5e03e2de28989493e0c48.png)
![homepage.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/5b3fa152acf9c70015453b07/125516c66e82c728ace21e0d46e09b97269282ae8aabf301b44fff9e2c4533317346f6292380e779eb1a22005f561ff941585e16faf08e2da7b32e6c981e5ab38532022d8c6853beb2901eb80bbb258f4f80bebf8249c2ed3a2df5c6.png)

**Comment by cachemeoutside** [comment]
Based on the test for the search results, it looks like the images take an average of 269.75ms(~.270 seconds) to receive the images. Not sure if the application is loading the thumbnails sequentially or in parallel.

**Comment by ksclarke** [comment]
So, that looks to me like statuscake doesn't measure things loaded via AJAX. We can compare the list of files Mark's screenshot shows for the single item view and they're not in the screenshot Anthony shared of statuscake. There should be a bunch of default.jpg files, among other things.

**Comment by cachemeoutside** [comment]
@ksclarke yup, I don't think statuscake has an option to measure things loaded via AJAX. I believe those could be rendered during and even after the page is fully loaded. I'll inquire with Statuscake to see if there's a solution.

**Comment by cachemeoutside** [comment]
There doesn't seem to be a visible way to create a public dashboard to view page speed tests. I've inquired more details from statuscake(email shown below), but granted view access to the following emails:

- mark@curationexperts.com
- bess@curationexperts.com

@bess, @mark-dce, when you get a chance, please login and confirm that you have access to statuscake. You should've received an invitation to create a subaccount. You may have to relogin after creating your account(statuscake is a bit wonky).

If there are any other accounts to be added, please let me know.

Email template sent to statuscake:
--
Hello, I have a few questions:
1. Regarding subaccounts: I've noticed that View privileged accounts are able to edit the Page Speed Monitoring tests, is this behaviour expected?
2. Tags are not available in Page Speed Monitoring tests, will this be implemented in the future?
3. For page speed monitoring tests, we've noticed that it doesn't track AJAX load times. Is there a way to enable this?
Thanks,
- Anthony

**Comment by cachemeoutside** [comment]
Looks like Statuscake does not have a way to test AJAX load times. Snippet below.

```
Sorry but due to the AJAX being triggered after the full load event it's not possible for us to test this and still give a realistic full load time. 
```

**Comment by ksclarke** [comment]
@cachemeoutside I'd be interested in statuscake access too so I can see test results.

**Comment by cachemeoutside** [comment]
@ksclarke Invite sent to your library email. You *will* have to relogin after creating your account or else you won't see anything.

**Comment by cachemeoutside** [comment]
@bess  and I have started a confluence page to document our fixed load times test. Chrome developer tools are used to test AJAX content, Statuscake will supplement and be used to track static file load times.

https://docs.library.ucla.edu/display/CUP/Load+Time+Datapoints
